src/cargo/ops/cargo_new.rs

```diff
@@ -444,7 +444,7 @@ fn mk(config: &Config, opts: &MkOptions) -> CargoResult<()> {
 
     match vcs {
         VersionControl::Git => {
-            if !fs::metadata(&path.join(".git")).is_ok() {
+            if fs::metadata(&path.join(".git")).is_err() {
```

Comment:
!.is_ok reads somewhat better to me...

Maybe let’s make it radically better and extract a 

    let exists = |dotdir: &str| -> bool

function while we are at it?

Reply:
@alexcrichton looks like we use to ways to check if a file exists in cargo:

* `Path.exists()`
* `fs::metadata()::is_ok()`

Which one is preferred and why? :)

Reply:
Oh I'd prefer `exists` today. Usage of `metadata` predates the existence of `exists`